src/Cake.Core/IO/IFileSystemInfo.cs

```diff
@@ -30,5 +32,11 @@ public interface IFileSystemInfo
         ///   <c>true</c> if the entry is hidden; otherwise, <c>false</c>.
         /// </value>
         bool Hidden { get; }
+
+        /// <summary>
+        /// Gets the time when this <see cref="IFileSystemInfo"/> was last written to.
+        /// </summary>
+        /// <value>The last write time.</value>
+        DateTime LastWriteTime { get; }
```

Comment:
@cake-build/cake-team please review breaking change.

Reply:
Is there any way we could get LastWriteTime without breaking?

Reply:
Yeah I can't see a way to do this without a breaking change; perhaps C# 8 would help in future.

FWIW, all the existing implementations (`Directory`, `File`, `FakeDirectory` and `FakeFile`) are `sealed` and I implemented those.

Having access to directory and file timestamps is really useful so could we just push it through and manage it like previous breaking changes?

Reply:
We're in general restrictive with breaking changes, and always try to weigh benefit vs. risk, and if possible see if we can avoid. Breaking changes means all existing addins/modules/assemblies will need to be updated, recompiled and published. Sometimes reward justifies it, but it also means we'll review and discuss it more before merging. 

We've added he notion of possible breaking change, when we think it's a low risk, we will warn but still execute, if we know it's a high impact breaking change we'll not even load modules/addins targeting previous version.

On the module side personally I only know of one that's implementing IFileSystemInfo / IFile / IDirectorym so might be a low risk but we would need to test and inventory that to deem low risk.

Reply:
Okay, it sounds like it might be low risk as it will likely only affect modules. Is there a list of publicly available modules somewhere? How can I help?

Reply:
Meanwhile, I've just realised there's a more elegant solution: remove the breaking change to `IFileSystemInfo`, keep my changes to the existing implementations (`Directory`, `File`, `FakeDirectory` and `FakeFile`), and downcast types for timestamps in `Zipper`.

Reply:
See commit [3039e28c](https://github.com/cake-build/cake/pull/2382/commits/3039e28c6c5f8b9eb6e923c6d543bbaec1e041e4).

The Azure Pipelines build for `Build Cake Windows` had an unrelated failure. Is there a way to retry those builds without another commit?

Reply:
I've queued a new build.

Reply:
GH status not reflecting it but build succeded
https://dev.azure.com/cake-build/Cake/_build/results?buildId=2690